File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch02/C/P02-Swap.cpp
// Returning swapped arguments from a function (C++ syntax)
#include <stdio.h>

void Swap(double &x, double &y)              // arguments passed by reference
{
   double temp;                                         // temporary variable

   temp = x; x = y; y = temp;
}

void main()
{
   double a, b;

   a = 1e0; b = 2e0;                                  // values to be swapped
   Swap(a,b);
   printf("%f %f\n",a,b);
}

File: Introduction_to_numerical_programming_using_Python_and_CPP_Beu/Ch02/Python/P02-Swap.py
# Returning swapped arguments from a function

def Swap(x, y):                            # scalar arguments passed by value
   temp = x; x = y; y = temp
   return (x, y)                                 # return interchanged values

# main

a = 1e0; b = 2e0                                       # values to be swapped
(a,b) = Swap(a,b)      # left side - return values; right side - input values
print(a,b)
